server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2200,11 +2177,7 @@ private IndexWriter createWriter() throws IOException {
 
     // pkg-private for testing
     IndexWriter createWriter(Directory directory, IndexWriterConfig iwc) throws IOException {
-        if (Assertions.ENABLED) {
-            return new AssertingIndexWriter(directory, iwc);
```

Comment:
why not keep `AssertingIndexWriter`? Perhaps still useful to check that updateDocument(s) / deleteDocuments / tryDeleteDocument is never called?

Reply:
ok, restored in https://github.com/elastic/elasticsearch/pull/50415/commits/60139407229a51c670a67a4f594e4b239b2d3e04.